libs/poses/include/mrpt/poses/gtsam_wrappers.h

```diff
@@ -50,6 +52,56 @@ static mrpt::math::TPose3D toTPose3D(const gtsam::Pose3& p)
 	return mrpt::poses::CPose3D(H).asTPose();
 }
 
+static void jacobian_rodrigues_from_YPR(
```

Comment:
Same here with these two jacobian functions... Perhaps they could be moved to CPose3D.h as members returning the matrices by value?

Remember that in this case, it would be preferred to just declare the methods in the CPose3D.h file, then implement them in the CPose3D.cpp file. 

There, you are free to use `Eigen::*` classes. In the case of `jacobian_rodrigues_from_quat()`, since CQuaternion is a template, the whole thing would go in the .h, so the only direct reference to Eigen stuff (`(quaternion.asEigen().isUnitary());`)  should be replaced by `ASSERT_NEAR_(quaternion.normSqr(), 1.0, 1e-5);` or alike. 

Note that Eigen headers are not directly included in any MRPT public header since they introduce a huge penalty in compilation time... sorry, it's quite complicated,I know! ;-)

Reply:
OK. I hope I got it right :)
Regarding the asserts, I can also delete them if you think they should not be there

Reply:
In `poses/gtsam_wrapper.h` I left the `.asEigen()` in line 64 because I saw it was also in line 44, but I can change it if needed.